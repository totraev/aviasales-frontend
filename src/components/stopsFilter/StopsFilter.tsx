import React, { SFC } from 'react';
import styles from './StopsFilter.module.css';

import Checkbox from '../checkbox/Checkbox';

const StopsFilter: SFC = () => (
  <div className={styles.stops}>
    <Checkbox />
    <Checkbox />
    <Checkbox />
    <Checkbox />
  </div>
);

export default StopsFilter;
