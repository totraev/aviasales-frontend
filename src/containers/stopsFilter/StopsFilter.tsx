import React, { SFC } from 'react';
import styles from './StopsFilter.module.css';

import { compose, withHandlers } from 'recompose';

import Checkbox from '../../components/checkbox/Checkbox';
import { IHandlers, IInnerProps, IStopsFilterProps } from './StopsFilter.types';

export const StopsFilter: SFC<IInnerProps> = ({ stopsList, handleClick, handleUncheckOther }) => (
  <div className={styles.stops}>
    {stopsList.map((checked, i) => (
      <Checkbox
        id={`stops_${i}`}
        key={i}
        stops={i}
        value={checked}
        onClick={handleClick(i)}
        onUncheckOther={handleUncheckOther(i)}
      />
    ))}
  </div>
);

export default compose<IInnerProps, IStopsFilterProps>(
  withHandlers<IStopsFilterProps, IHandlers>({
    handleClick: ({ stopsList }) => (stops) => () => {
      // tslint:disable-next-line
      console.log(stops, stopsList[stops]);
    },
    handleUncheckOther: ({ stopsList }) => (stops) => () => {
      // tslint:disable-next-line
      console.log(stops, stopsList[stops]);
    },
  }),
)(StopsFilter);
