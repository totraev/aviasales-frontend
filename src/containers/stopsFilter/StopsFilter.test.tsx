import React from 'react';
import renderer from 'react-test-renderer';

import { StopsFilter } from './StopsFilter';

describe('StopsFilter component', () => {
  it('Should renders correctly', () => {
    const mockFunc: any = () => null;
    const tree = renderer.create(
      <StopsFilter
        stopsList={[true, false, false, true]}
        handleClick={mockFunc}
        handleUncheckOther={mockFunc}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
