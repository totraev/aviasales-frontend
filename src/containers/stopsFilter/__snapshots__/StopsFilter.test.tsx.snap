// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopsFilter component Should renders correctly 1`] = `
<div
  className="stops"
>
  <div
    className="wrap"
  >
    <label
      className="label"
      htmlFor="stops_0"
    >
      <input
        checked={true}
        className="checkbox"
        id="stops_0"
        onClick={null}
        type="checkbox"
      />
      <span
        className="faceChecked"
      />
      БЕЗ ПЕРЕСАДОК
    </label>
    <div
      className="append"
    >
      <button
        className="uncheckOther"
        onClick={null}
      >
        ТОЛЬКО
      </button>
    </div>
  </div>
  <div
    className="wrap"
  >
    <label
      className="label"
      htmlFor="stops_1"
    >
      <input
        checked={false}
        className="checkbox"
        id="stops_1"
        onClick={null}
        type="checkbox"
      />
      <span
        className="faceChecked"
      />
      1 ПЕРЕСАДКА
    </label>
    <div
      className="append"
    >
      <button
        className="uncheckOther"
        onClick={null}
      >
        ТОЛЬКО
      </button>
    </div>
  </div>
  <div
    className="wrap"
  >
    <label
      className="label"
      htmlFor="stops_2"
    >
      <input
        checked={false}
        className="checkbox"
        id="stops_2"
        onClick={null}
        type="checkbox"
      />
      <span
        className="faceChecked"
      />
      2 ПЕРЕСАДКИ
    </label>
    <div
      className="append"
    >
      <button
        className="uncheckOther"
        onClick={null}
      >
        ТОЛЬКО
      </button>
    </div>
  </div>
  <div
    className="wrap"
  >
    <label
      className="label"
      htmlFor="stops_3"
    >
      <input
        checked={true}
        className="checkbox"
        id="stops_3"
        onClick={null}
        type="checkbox"
      />
      <span
        className="faceChecked"
      />
      3 ПЕРЕСАДКИ
    </label>
    <div
      className="append"
    >
      <button
        className="uncheckOther"
        onClick={null}
      >
        ТОЛЬКО
      </button>
    </div>
  </div>
</div>
`;
